crates/services/src/service.rs

```diff
@@ -101,16 +101,25 @@ pub trait RunnableTask: Send {
 #[derive(Debug)]
 pub struct ServiceRunner<S>
 where
-    S: RunnableService,
+    S: RunnableService + 'static,
 {
     /// The shared state of the service
     pub shared: S::SharedData,
     state: Shared<watch::Sender<State>>,
 }
 
+impl<S> Drop for ServiceRunner<S>
+where
+    S: RunnableService + 'static,
+{
+    fn drop(&mut self) {
+        self.stop();
+    }
+}
+
 impl<S> Clone for ServiceRunner<S>
 where
-    S: RunnableService,
+    S: RunnableService + 'static,
 {
     fn clone(&self) -> Self {
```

Comment:
We should either remove clone or make drop only stop the service when the last cloned instance of the service runner drops.